src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -842,11 +842,7 @@ public string RequestLicKey(Type type)
 
             ((IDisposable?)parameters[2])?.Dispose();
 
-            var licenseKey = (string?)parameters[3];
-            if (licenseKey == null)
-            {
-                throw new COMException(); // E_FAIL
-            }
+            var licenseKey = (string?)parameters[3] ?? throw new COMException(); // E_FAIL
```

Comment:
```suggestion
            return parameters[3] as string ?? throw new COMException(); // E_FAIL
```

Reply:
Why? You're concerned `parameters[3]` may be something other than `null` or `string`, or you're just trying to avoid the `?`?

Reply:
It's a simplification of behaviour, instead of potentially throwing `InvalidCastException`, just throw `COMException` upon failure.